independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/ModelUtils.java

```diff
@@ -232,7 +232,9 @@ private static Properties loadPomProps(Path appJar, Path artifactIdPath) throws
     }
 
     public static Model readModel(final Path pomXml) throws IOException {
-        return readModel(Files.newInputStream(pomXml));
+        Model model = readModel(Files.newInputStream(pomXml));
+        model.setPomFile(pomXml.toFile());
+        return model;
     }
```

Comment:
I'm not sure I understand this change and how it could affect adding an extension? I thought this would be used only for the bootstrap and that we were using other calls for actually manipulating the pom file?

I suppose we missed something but it would be good to clarify what.

Reply:
Wouldn't the right fix be to avoid calling the bootstrap `ModelUtils.readModel()` in `MavenProjectBuildFile`? I would expect us to use the formatting-aware one here?

My guess is that it would fix the issue and be more future proof.

That being said, don't revert the change you made as I think it's good to do it anyway.

Reply:
@gsmet the Model that is passed to the `quarkus extension add` is read here (bootstrap). When persisting the model the maven-model-helper library is called, but because it requires a path to the original file to compare with the model to be persisted, you need to set this attribute (you could also pass the path to the file when persisting but that would impact on the existing public method signature in Quarkus core). This is done in the library here: https://github.com/fabric8-launcher/maven-model-helper/blob/main/src/main/java/io/fabric8/maven/Maven.java#L66, but because the change was reverted, we need to explicitly invoke it

Reply:
> Wouldn't the right fix be to avoid calling the bootstrap `ModelUtils.readModel()` in `MavenProjectBuildFile`? I would expect us to use the formatting-aware one here?

There is no need to change that now. The API to read the model is the same (except that the `setPomFile` isn't invoked). Formatting only applies when persisting so we're good

Reply:
> the Model that is passed to the quarkus extension add is read here (bootstrap).

Read where? I might have missed something (it's late for me and the day has been quite full) but from what I can see, when executing the installation plan, we do it in `MavenProjectBuildFile`. And that file is not part of the bootstrap?

> There is no need to change that now. The API to read the model is the same (except that the setPomFile isn't invoked). Formatting only applies when persisting so we're good

But do we have good reasons to not be consistent and use the same API for reading and writing?
If we consume the model from the bootstrap then fine. But from what I can see, we are reading it specifically for executing the installation plan.

(Don't hesitate to push back, again, it's late and I'm tired)

Reply:
I'm not in my computer now (replying from mobile), but I'll confirm tomorrow

Reply:
@gsmet I've now changed the `MavenProjectBuildFile` to `MojoUtils.readPom`.